src/highlighting.js

```diff
@@ -9,23 +9,51 @@ function anyRegexesMatch(regexes, criteria) {
   return _.some(_.invokeMap(_.map(regexes, regexify), 'test', criteria))
 }
 
+// Convert the fields array to object map where we only pick the first key from the objects
+// Highlight fields can be either strings or objects with a single key which value is the ES highlights object config
+// If the highlight field is specific as a string only then it uses the default highlights config
+function arrayToHighlightsFieldMap(array) {
+  return _.reduce(
+    array,
+    (a, c) => {
+      if (_.isPlainObject(c)) {
+        let key = _.findKey(c, _.identity) // get the first key
+        a[key] = c[key]
+      } else {
+        a[c] = {}
+      }
+      return a
+    },
+    {}
+  )
+}
```

Comment:
using lodash/fp:

```js
_.flow(
  _.map(F.when(_.isString, x => ({ [x]: {} }))),
  _.mergeAll
)
```

https://runkit.com/daedalus28/refactored-arraytohighlightsfieldmap

Reply:
Sure. We can add `F.ifElse(_.isEmpty, _.always({}), _.mergeAll)` since in the case of an empty array it does not match the previous version in terms of output